### PR TITLE
Update SchemaReflector to return all transitive deps

### PR DESCRIPTION
**Pull request by juliaogris**
Update SchemaReflector to return all transitive dependencies of
underlying .proto file, i.e. all FileDescriptors of all imported .proto
files and their imports transitively.

This update complies with the gRPC server reflection spec:

> Because most usecases will require also requesting the transitive
> dependencies of requested files, the queries will also return all
> transitive dependencies of the returned file. Should interesting
> usecases for non-transitive queries turn up later, we can easily extend
> the protocol to support them.

https://github.com/grpc/grpc/blob/master/doc/server-reflection.md#protobuf-descriptors

### CODE REVIEW

File: wire-library/wire-reflector/src/main/kotlin/com/squareup/wire/reflector/SchemaReflector.kt [hunk 1]
```diff
@@ -135,9 +137,10 @@ class SchemaReflector(
 
   private fun fileByFilename(request: ServerReflectionRequest): ServerReflectionResponse {
     val protoFile = schema.protoFile(request.file_by_filename!!)!!
-
+    val allProtoFiles = allDependencies(protoFile)
+    val schemaEncoder = SchemaEncoder(schema)
     return ServerReflectionResponse(
-      file_descriptor_response = SchemaEncoder(schema).encode(protoFile).toFileDescriptorResponse()
+            file_descriptor_response = FileDescriptorResponse(allProtoFiles.map { schemaEncoder.encode(it) })
```
Comment (chris-ryan-square) [inline wire-library/wire-reflector/src/main/kotlin/com/squareup/wire/reflector/SchemaReflector.kt:143]: nit: usually 2 space indent.

File: wire-library/wire-reflector/src/main/kotlin/com/squareup/wire/reflector/SchemaReflector.kt [hunk 2]
```diff
@@ -153,9 +156,10 @@ class SchemaReflector(
       )
 
     val protoFile = schema.protoFile(location.path)!!
-    val result = SchemaEncoder(schema).encode(protoFile).toFileDescriptorResponse()
+    val allProtoFiles = allDependencies(protoFile)
+    val schemaEncoder = SchemaEncoder(schema)
     return ServerReflectionResponse(
-      file_descriptor_response = result
+            file_descriptor_response = FileDescriptorResponse(allProtoFiles.map { schemaEncoder.encode(it) })
```
Comment (chris-ryan-square) [inline wire-library/wire-reflector/src/main/kotlin/com/squareup/wire/reflector/SchemaReflector.kt:162]: nit: indent

File: wire-library/wire-reflector/src/main/kotlin/com/squareup/wire/reflector/SchemaReflector.kt (hunk 1 above)
Comment (juliaogris) [reply wire-library/wire-reflector/src/main/kotlin/com/squareup/wire/reflector/SchemaReflector.kt]: fixed, intelliJ did for me!?!

File: wire-library/wire-reflector/src/main/kotlin/com/squareup/wire/reflector/SchemaReflector.kt [hunk 3]
```diff
@@ -187,4 +191,25 @@ class SchemaReflector(
       file_descriptor_proto = listOf(this)
     )
   }
+
+  // allDependencies returns protoFile and all its transitive dependencies in topographical order such that files always
+  // appear after their dependencies.
```
Comment (swankjesse) [inline wire-library/wire-reflector/src/main/kotlin/com/squareup/wire/reflector/SchemaReflector.kt:196]: Ooooh good comment. Knowing it's bottom up is handy.

File: wire-library/wire-schema/build.gradle.kts [hunk 4]
```diff
@@ -74,3 +74,6 @@ configure<MavenPublishBaseExtension> {
     KotlinMultiplatform(javadocJar = Dokka("dokkaGfm"))
   )
 }
+
+tasks.withType<ProcessResources> { duplicatesStrategy = DuplicatesStrategy.INCLUDE}
```
Comment (swankjesse) [inline wire-library/wire-schema/build.gradle.kts:78]: I wonder if this is strictly necessary... I saw that the build failed without it, but we're not changing anything here that would trigger this.

**Review by swankjesse — APPROVED** [review]
Nice PR.

**Pull request merged**